openpype/hosts/blender/plugins/create/create_blendScene.py

```diff
@@ -31,21 +31,11 @@ def _process(self):
         asset = self.data["asset"]
         subset = self.data["subset"]
         name = plugin.asset_name(asset, subset)
-        asset_group = bpy.data.objects.new(name=name, object_data=None)
-        asset_group.empty_display_type = 'SINGLE_ARROW'
-        instances.objects.link(asset_group)
+
+        # Create the new asset group as collection
+        asset_group = bpy.data.collections.new(name=name)
+        instances.children.link(asset_group)
         self.data['task'] = get_current_task_name()
         lib.imprint(asset_group, self.data)
 
-        # Add selected objects to instance
-        if (self.options or {}).get("useSelection"):
-            bpy.context.view_layer.objects.active = asset_group
-            selected = lib.get_selection()
-            for obj in selected:
-                if obj.parent in selected:
-                    obj.select_set(False)
-                    continue
-            selected.append(asset_group)
-            bpy.ops.object.parent_set(keep_transform=True)
```

Comment:
Shouldn't we still be linking the selection has children of the collection? (not reparenting but adding it to the collection on create?)

Reply:
Yes, but Blender doesn't allow to get the selected collections :/ if we allow to add the selection and add only the objects, I'm afraid it would be more complicated later to add what the artists want (because if they added, for instance, an object that was inside the collection, they need to unlink the object and link the collection). Instead, if we don't add anything, they just need to link what they need. It's a compromise... but I think it's the best way, until they allow to get the selected collections at least.

Reply:
Would [this](https://devtalk.blender.org/t/how-can-i-get-the-collection-that-is-selected-in-the-outliner-in-python/16572/5) work?

Reply:
You know, I've tried a lot of those that you linked, but I didn't find this: [blender.stackexchange.com/a/299669](https://blender.stackexchange.com/a/299669) and... it works 😳
I had the same problem of the OP of that question, I wasn't able to make `bpy.context.selected_ids` run in 3.6, and seriously thought that they removed it without updating the docs (it happened before 😅)!

So thanks, I'll change it to add it! 😄

Reply:
So, update about this! After various tests, I've found that the legacy creator uses a method  called `maintained_selection`, which "resets" the selected objects. This means that it deselects everything, and selects again the original selection. The problem is, that this deselects the selected collections, and while we found a way to get it, there is no way to select them back 😅 or at least, I haven't found it. I've looked extensively, but couldn't find anything 🥲
I don't think that `maintained_selection` is used in the new publisher's creator, so at least we can fix it when we implement the new publisher in Blender, but unless we find a way around it, for now I don't think we can add the selection.

Reply:
I think since this will only happen on Create shouldn't we be fine?

Reply:
Nevermind, I'm just stupid and I just noticed that I can disable the `maintain_selection` 🤦 intense week 🥲